[ML] Fix values for decision path not showing correctly for regression due to rounding

qn895 opened the pull request:
## Summary

This PR fixes an formatting issue which will rounds large numbers (e.g. 1323423 to 1320000). It now uses `formatSingleValue` to show up to 3 significant figures for small numbers, and to round up to first decimal for large numbers.

- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

qn895 committed:
[ML] Change to formatSingleValue for tickFormatter

qn895 committed:
[ML] Change to formatSingleValue for tickFormatter

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

smith committed:
Set new data on elements after fetch. (#80430)

When `add`ing an element with an already existing ID, it will use the already existing data instead of setting new data. This made it so when you change the time picker the element health statuses would not update.

When we get new elements, replace the existing data for existing elements.

Fixes #80335.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/)
* Commit: 6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f) | [after](https://github.com/elastic/kibana/commit/396135a05d00127bfd459cc50ad5b592d8b00027) | diff |
| --- | --- | --- | --- |
| `ml` | 11.3MB | 11.3MB | +5.9KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/","number":"81500","commit":"6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f"}]}
PIPELINE-->

renovate[bot] committed:
Update dependency vega-lite to ^4.17.0 (#80382)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

pheyos committed:
[ML] Functional tests - stabilize calendar service methods (#80349)

This PR stabilizes the functional calendar tests, which are flaky on cloud.

pgayvallet committed:
Use `SavedObjectClass` from `savedObject` start contract instead of `createSavedObjectClass` (#80063)

* no longer export static createSavedObjectClass from savedObject plugin

* fix forgotten call

* yet another usage

* more fixes

* move so to required plugins for timelion

VladLasitsa committed:
Filter upper range not shown in filter edit UI (#79683)

* use 'lte' if 'lt' is undefined

* Fix unit test

* Update src/plugins/data/common/es_query/filters/get_filter_params.ts

Co-authored-by: Tim Roes <mail@timroes.de>

Co-authored-by: Tim Roes <mail@timroes.de>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
respect requested sub vis type for xy charts (#80297)

peteharverson committed:
[ML] Fix Anomaly Explorer charts time range to obey time picker range (#80317)

* [ML] Fix Anomaly Explorer charts time range to obey time picker range

* [ML] Fix explorer_charts_container_service Jest tests

stratoula committed:
[TSVB] Display epoch on a more human readable format (#79110)

* [TSVB] Display epoch on a more human readable format

* fix test

* Fix date tests

* Nit on the formatter function

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

pheyos committed:
[ML] Skip cardinality validation API test

Commit:
[Ingest Manager] Add .catch to handle Promise rejection (#80056)

## Summary

Prevent an unhandled rejection (which crashes Kibana) when installing a package. Initially found as a part of https://github.com/elastic/kibana/issues/79560#issuecomment-704581602 & [this PR](https://github.com/elastic/kibana/pull/79791/files#diff-e2fac97e334155d887ecee2a8434120f9150f65c85725c4a25f261da79233cf0R336-R340) to fix it. 

While the conditions of https://github.com/elastic/kibana/issues/79560 are a bit of an edge case, we could encounter this behavior on any expected error in this promise chain.

 * Add .catch to prevent unhandled promise rejection warning/crash.
 * pulled `installPackage` out to a separate file so it's easier to test
 * added a test for this specific error where an error in `installIndexPatterns` or `installKibanaAssets` would ultimately crash Kibana. More detail in the comments https://github.com/elastic/kibana/blob/608fc3ef1b1306fc0dd76ff07b6652c221ed242b/x-pack/plugins/ingest_manager/server/services/epm/packages/_install_package.ts#L81-L95



### Checklist
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

flash1293 committed:
Allow sorting by median (#79839)

pheyos committed:
[ML] Skip cardinality validation API test part 2

stratoula committed:
[Discover] Skip the Discover to Visualize functional tests for cloud (#80469)

walterra committed:
[ML] Fix color coding for outlier detection. Fix nested fields for data gird. (#80467)

- Fix for color coding for outlier detection due to updated feature influence format.
- Fix for displaying cell values for nested fields.

flash1293 committed:
[Lens] Custom labels for ranges (#79628)

Commit:
[Ingest Manager] Don't error if updating an agent policy (#80403)

## Summary

fixes #80136

Haven't added/updated tests yet but tested the various combinations in the UI and all worked as expected.

<details><summary>Screenshots</summary>
<h3>Policy listing</h3>
<img width="1302" alt="Screen Shot 2020-10-14 at 7 05 19 AM" src="https://user-images.githubusercontent.com/57655/95981056-1c1c6d00-0dec-11eb-9746-5e91043f3d19.png">

<h3>Update existing policy succeeds</h3>
<img width="1308" alt="Screen Shot 2020-10-14 at 7 05 49 AM" src="https://user-images.githubusercontent.com/57655/95981059-1cb50380-0dec-11eb-9e55-7a14b72085f9.png">
<h3>Creating new policy w/existing name fails</h3>
<img width="1308" alt="Screen Shot 2020-10-14 at 7 06 06 AM" src="https://user-images.githubusercontent.com/57655/95981062-1cb50380-0dec-11eb-833b-1ab6636a5c67.png">

<h3>Updating policy to existing name fails</h3>
<img width="1308" alt="Screen Shot 2020-10-14 at 7 08 05 AM" src="https://user-images.githubusercontent.com/57655/95981078-1f175d80-0dec-11eb-91d5-7ffd976da48e.png">

</details>
### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

FrankHassanabad committed:
[Security Solutions][Detection Engine] Fixes critical bug where an invalid/empty mapping can cause a match all (#80415)

## Summary

Critical bug where if you have an invalid/empty mapping this can cause the threat match to end up matching against everything rather than matching against an empty set.

Added integration end to end tests to exercise both this case and the positive case where regular mappings should match against 10+ things.

If you want to test this manually you can do the following:

First find a field that doesn't have any data in it such as `url.extension` using timeline

<img width="1160" alt="Screen Shot 2020-10-13 at 2 54 17 PM" src="https://user-images.githubusercontent.com/1151048/95915285-2c890500-0d64-11eb-8a78-3a12caf280f5.png">

Then create the rule with it using any other matching field:
<img width="1036" alt="Screen Shot 2020-10-13 at 2 53 39 PM" src="https://user-images.githubusercontent.com/1151048/95915319-36ab0380-0d64-11eb-9d54-65978279586c.png">

Then run the rule and see that you have signals when there should be zero signals happening.
 
### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

lizozom committed:
Speed up filter creation in functional tests (#80037)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

VladLasitsa committed:
Timelion y-axis label not displayed (#80007)

* Moved flot in ui-shared-deps and leavt only one version.

* Removed unneeded import

* use commonJS so we can execute code between loading specific modules

* prefix i18n keys with flot.

* Fix Internationalization

* remove duplicate pie plugin

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[kbn/std] update rxjs helpers to use simple operators (#80416)

Co-authored-by: spalger <spalger@users.noreply.github.com>

peteharverson commented on x-pack/plugins/ml/public/application/formatters/format_value.ts:
```diff
@@ -53,7 +53,7 @@ export function formatValue(
 // containing the timestamp of the anomaly should be supplied in
 // order to correctly format the day or week offset to the time of the anomaly.
-function formatSingleValue(
+export function formatSingleValue(
```
Maybe you should change the `mlFunction` to be optional here? Originally this was written for formatting values from anomaly records, but this is a good example of where the function has uses outside of anomaly records. Then you wouldn't need to pass the empty string when calling it from the decision path chart.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/80448/commits/4c3996242651e19ece5c0ac291743f3d012b24de

chrisronline committed:
[Monitoring] Ensure alerting is optional (#79168)

* Ensure alerting is optional

* Fix the naming

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

peteharverson reviewed (approved):
Tested and LGTM. Left one minor suggestion.

pzl committed:
[Security Solution] Use Agent.id for endpoint pivot ID (#74272)

* switch endpoint meta query to use agent.id

* update policy route to use agent ID

* update policy unit test, with schema change

* security front-end use agent.id as identifier

* update test to check the right field

* update SIEM to get endpoint data by agent.id

* fix type in test, but need to fix data, will still fail

* test: pull agent ID from esarchive data

* magnets, how do they work?

* cleanup

* apparently this test works differently now

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
Agent policy names must be unique across Fleet (#80506)

qn895 committed:
[ML] Make formatSingleValue mlfunction optional

qn895 committed:
[ML] Make formatSingleValue mlfunction optional

spalger committed:
skip flaky suite (#80104)

szabosteve committed:
[DOCS] Adds initial content to ML readme.md (#80478)

Commit:
[kbn/optimizer] add support for --focus option (#80436)

Co-authored-by: spalger <spalger@users.noreply.github.com>

andrew-goldstein committed:
Fixes an issue where zoom controls were overlaid by the Timeline flyout button when "analyze event" is in full screen mode (#80429)

## Fixes an issue where zoom controls were overlaid by the Timeline flyout button when "analyze event" is in full screen mode

This PR implements a fix for [Zoom-in Zoom-out widget is overlaid on the Timeline flyout button after clicking on full-screen icon under "analyze event" #80309](https://github.com/elastic/kibana/issues/80309)

**Steps to Reproduce**

From the issue above:

1. Navigate to Detections under the security tab.
2. Click on the "analyze event" icon of alert that triggered in preconditions under the alert table.
3. Now click on the "Fullscreen" icon.
4. Observe that the Zoom-in Zoom-out widget is overlaid on the Timeline flyout button after clicking on full-screen icon under "analyze event". 

**Actual Result**
Zoom-in Zoom-out widget is overlaid on the Timeline flyout button after clicking on full-screen icon under "analyze event".

![UI_overlaid_events](https://user-images.githubusercontent.com/61860752/95853499-277f8200-0d73-11eb-8d17-6da7e37b0b14.jpg)

**Expected Result**
Zoom-in Zoom-out widget should not be overlaid on Timeline flyout button after clicking on full screen icon under "analyze event".

Screenshots of the fix:

### Chrome `85.0.4183.121` 

<img width="1708" alt="after-chrome" src="https://user-images.githubusercontent.com/4459398/95923597-13d41b80-0d73-11eb-9447-6ffcaeb12360.png">

### Firefox `81.0.1 `

<img width="1708" alt="safari-after" src="https://user-images.githubusercontent.com/4459398/95923622-24849180-0d73-11eb-859d-31c654588072.png">

### Safari `14.0`

<img width="1708" alt="safari-after" src="https://user-images.githubusercontent.com/4459398/95923656-36fecb00-0d73-11eb-881e-531208cce5fc.png">

spalger committed:
fix plugin-list.asciidoc

stratoula committed:
Register the Visualize Field Action of OSS on plugin setup (#80476)

dhurley14 committed:
[Security Solution] [Detections] useMemo for conditional rendering of actions step and bug fix (#80361)

* useMemo for conditional rendering of actions components

* fixes bug where state of form was being updated before components were mounted when switching between tabs on edit rule

XavierM committed:
[SECURITY SOLUTION] Bring back splash screen (#80404)

* bring back splash screen

* add tests

* add cypress test

alvarezmelissa87 reviewed (approved):
LGTM ⚡

qn895 committed:
Merge branch 'ml-fix-formatting-decision-path-incorrect-for-large-num' of https://github.com/qn895/kibana into ml-fix-formatting-decision-path-incorrect-for-large-num

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/)
* Commit: 4c3996242651e19ece5c0ac291743f3d012b24de
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/execution/node/628/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/execution/node/731/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/testReport)

<details><summary>X-Pack API Integration Tests.x-pack/test/api_integration/apis/ml/job_validation/cardinality·ts.apis Machine Learning job validation ValidateCardinality should recognize a high model plot cardinality</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_ml_job_validation_cardinality·ts/apis_Machine_Learning_job_validation_ValidateCardinality_should_recognize_a_high_model_plot_cardinality)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:07:09]           └-: Machine Learning
[00:07:09]             └-> "before all" hook
[00:07:09]             └-> "before all" hook
[00:07:09]               │ debg creating role ft_ml_source
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_source]
[00:07:09]               │ debg creating role ft_ml_source_readonly
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_source_readonly]
[00:07:09]               │ debg creating role ft_ml_dest
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_dest]
[00:07:09]               │ debg creating role ft_ml_dest_readonly
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_dest_readonly]
[00:07:09]               │ debg creating role ft_ml_ui_extras
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_ui_extras]
[00:07:09]               │ debg creating role ft_default_space_ml_all
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_all]
[00:07:09]               │ debg creating role ft_default_space_ml_read
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_read]
[00:07:09]               │ debg creating role ft_default_space_ml_none
[00:07:09]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_none]
[00:07:09]               │ debg creating user ft_ml_poweruser
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_poweruser]
[00:07:10]               │ debg created user ft_ml_poweruser
[00:07:10]               │ debg creating user ft_ml_poweruser_spaces
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_poweruser_spaces]
[00:07:10]               │ debg created user ft_ml_poweruser_spaces
[00:07:10]               │ debg creating user ft_ml_viewer
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_viewer]
[00:07:10]               │ debg created user ft_ml_viewer
[00:07:10]               │ debg creating user ft_ml_viewer_spaces
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_viewer_spaces]
[00:07:10]               │ debg created user ft_ml_viewer_spaces
[00:07:10]               │ debg creating user ft_ml_unauthorized
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_unauthorized]
[00:07:10]               │ debg created user ft_ml_unauthorized
[00:07:10]               │ debg creating user ft_ml_unauthorized_spaces
[00:07:10]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_unauthorized_spaces]
[00:07:10]               │ debg created user ft_ml_unauthorized_spaces
[00:08:46]             └-: job validation
[00:08:46]               └-> "before all" hook
[00:08:48]               └-: ValidateCardinality
[00:08:48]                 └-> "before all" hook
[00:08:48]                 └-> "before all" hook
[00:08:48]                   │ info [ml/ecommerce] Loading "mappings.json"
[00:08:48]                   │ info [ml/ecommerce] Loading "data.json.gz"
[00:08:48]                   │ info [ml/ecommerce] Skipped restore for existing index "ft_ecommerce"
[00:08:49]                   │ debg applying update to kibana config: {"dateFormat:tz":"UTC"}
[00:08:49]                 └-> should recognize a valid cardinality
[00:08:49]                   └-> "before each" hook: global before each
[00:08:49]                   └- ✓ pass  (23ms) "apis Machine Learning job validation ValidateCardinality should recognize a valid cardinality"
[00:08:49]                 └-> should recognize a high model plot cardinality
[00:08:49]                   └-> "before each" hook: global before each
[00:08:49]                   └- ✖ fail: apis Machine Learning job validation ValidateCardinality should recognize a high model plot cardinality
[00:08:49]                   │       Error: expected [ { id: 'cardinality_model_plot_high',
[00:08:49]                   │     modelPlotCardinality: 4675 },
[00:08:49]                   │   { id: 'cardinality_partition_field',
[00:08:49]                   │     fieldName: 'order_id' } ] to sort of equal [ { id: 'cardinality_model_plot_high',
[00:08:49]                   │     modelPlotCardinality: 4711 },
[00:08:49]                   │   { id: 'cardinality_partition_field',
[00:08:49]                   │     fieldName: 'order_id' } ]
[00:08:49]                   │       + expected - actual
[00:08:49]                   │ 
[00:08:49]                   │        [
[00:08:49]                   │          {
[00:08:49]                   │            "id": "cardinality_model_plot_high"
[00:08:49]                   │       -    "modelPlotCardinality": 4675
[00:08:49]                   │       +    "modelPlotCardinality": 4711
[00:08:49]                   │          }
[00:08:49]                   │          {
[00:08:49]                   │            "fieldName": "order_id"
[00:08:49]                   │            "id": "cardinality_partition_field"
[00:08:49]                   │       
[00:08:49]                   │       at Assertion.assert (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:100:11)
[00:08:49]                   │       at Assertion.eql (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:244:8)
[00:08:49]                   │       at Context.it (test/api_integration/apis/ml/job_validation/cardinality.ts:97:23)
[00:08:49]                   │ 
[00:08:49]                   │ 

```

#### Stack Trace
```
{ Error: expected [ { id: 'cardinality_model_plot_high',
    modelPlotCardinality: 4675 },
  { id: 'cardinality_partition_field',
    fieldName: 'order_id' } ] to sort of equal [ { id: 'cardinality_model_plot_high',
    modelPlotCardinality: 4711 },
  { id: 'cardinality_partition_field',
    fieldName: 'order_id' } ]
    at Assertion.assert (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.eql (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:244:8)
    at Context.it (test/api_integration/apis/ml/job_validation/cardinality.ts:97:23)
  actual:
   '[\n  {\n    "id": "cardinality_model_plot_high"\n    "modelPlotCardinality": 4675\n  }\n  {\n    "fieldName": "order_id"\n    "id": "cardinality_partition_field"\n  }\n]',
  expected:
   '[\n  {\n    "id": "cardinality_model_plot_high"\n    "modelPlotCardinality": 4711\n  }\n  {\n    "fieldName": "order_id"\n    "id": "cardinality_partition_field"\n  }\n]',
  showDiff: true }
```
</details>

---

<details><summary>X-Pack API Integration Tests.x-pack/test/api_integration/apis/ml/job_validation/cardinality·ts.apis Machine Learning job validation ValidateCardinality should recognize a high model plot cardinality</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_ml_job_validation_cardinality·ts/apis_Machine_Learning_job_validation_ValidateCardinality_should_recognize_a_high_model_plot_cardinality_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:07:23]           └-: Machine Learning
[00:07:23]             └-> "before all" hook
[00:07:23]             └-> "before all" hook
[00:07:23]               │ debg creating role ft_ml_source
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_source]
[00:07:23]               │ debg creating role ft_ml_source_readonly
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_source_readonly]
[00:07:23]               │ debg creating role ft_ml_dest
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_dest]
[00:07:23]               │ debg creating role ft_ml_dest_readonly
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_dest_readonly]
[00:07:23]               │ debg creating role ft_ml_ui_extras
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_ml_ui_extras]
[00:07:23]               │ debg creating role ft_default_space_ml_all
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_all]
[00:07:23]               │ debg creating role ft_default_space_ml_read
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_read]
[00:07:23]               │ debg creating role ft_default_space_ml_none
[00:07:23]               │ info [o.e.x.s.a.r.TransportPutRoleAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added role [ft_default_space_ml_none]
[00:07:23]               │ debg creating user ft_ml_poweruser
[00:07:23]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_poweruser]
[00:07:23]               │ debg created user ft_ml_poweruser
[00:07:23]               │ debg creating user ft_ml_poweruser_spaces
[00:07:23]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_poweruser_spaces]
[00:07:23]               │ debg created user ft_ml_poweruser_spaces
[00:07:23]               │ debg creating user ft_ml_viewer
[00:07:23]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_viewer]
[00:07:23]               │ debg created user ft_ml_viewer
[00:07:23]               │ debg creating user ft_ml_viewer_spaces
[00:07:23]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_viewer_spaces]
[00:07:23]               │ debg created user ft_ml_viewer_spaces
[00:07:23]               │ debg creating user ft_ml_unauthorized
[00:07:23]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_unauthorized]
[00:07:23]               │ debg created user ft_ml_unauthorized
[00:07:23]               │ debg creating user ft_ml_unauthorized_spaces
[00:07:24]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-debian-tests-xxl-1602687241917693461] added user [ft_ml_unauthorized_spaces]
[00:07:24]               │ debg created user ft_ml_unauthorized_spaces
[00:09:05]             └-: job validation
[00:09:05]               └-> "before all" hook
[00:09:07]               └-: ValidateCardinality
[00:09:07]                 └-> "before all" hook
[00:09:07]                 └-> "before all" hook
[00:09:07]                   │ info [ml/ecommerce] Loading "mappings.json"
[00:09:07]                   │ info [ml/ecommerce] Loading "data.json.gz"
[00:09:07]                   │ info [ml/ecommerce] Skipped restore for existing index "ft_ecommerce"
[00:09:07]                   │ debg applying update to kibana config: {"dateFormat:tz":"UTC"}
[00:09:08]                 └-> should recognize a valid cardinality
[00:09:08]                   └-> "before each" hook: global before each
[00:09:08]                   └- ✓ pass  (16ms) "apis Machine Learning job validation ValidateCardinality should recognize a valid cardinality"
[00:09:08]                 └-> should recognize a high model plot cardinality
[00:09:08]                   └-> "before each" hook: global before each
[00:09:08]                   └- ✖ fail: apis Machine Learning job validation ValidateCardinality should recognize a high model plot cardinality
[00:09:08]                   │       Error: expected [ { id: 'cardinality_model_plot_high',
[00:09:08]                   │     modelPlotCardinality: 4675 },
[00:09:08]                   │   { id: 'cardinality_partition_field',
[00:09:08]                   │     fieldName: 'order_id' } ] to sort of equal [ { id: 'cardinality_model_plot_high',
[00:09:08]                   │     modelPlotCardinality: 4711 },
[00:09:08]                   │   { id: 'cardinality_partition_field',
[00:09:08]                   │     fieldName: 'order_id' } ]
[00:09:08]                   │       + expected - actual
[00:09:08]                   │ 
[00:09:08]                   │        [
[00:09:08]                   │          {
[00:09:08]                   │            "id": "cardinality_model_plot_high"
[00:09:08]                   │       -    "modelPlotCardinality": 4675
[00:09:08]                   │       +    "modelPlotCardinality": 4711
[00:09:08]                   │          }
[00:09:08]                   │          {
[00:09:08]                   │            "fieldName": "order_id"
[00:09:08]                   │            "id": "cardinality_partition_field"
[00:09:08]                   │       
[00:09:08]                   │       at Assertion.assert (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:100:11)
[00:09:08]                   │       at Assertion.eql (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:244:8)
[00:09:08]                   │       at Context.it (test/api_integration/apis/ml/job_validation/cardinality.ts:97:23)
[00:09:08]                   │ 
[00:09:08]                   │ 

```

#### Stack Trace
```
{ Error: expected [ { id: 'cardinality_model_plot_high',
    modelPlotCardinality: 4675 },
  { id: 'cardinality_partition_field',
    fieldName: 'order_id' } ] to sort of equal [ { id: 'cardinality_model_plot_high',
    modelPlotCardinality: 4711 },
  { id: 'cardinality_partition_field',
    fieldName: 'order_id' } ]
    at Assertion.assert (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.eql (/dev/shm/workspace/parallel/13/kibana/packages/kbn-expect/expect.js:244:8)
    at Context.it (test/api_integration/apis/ml/job_validation/cardinality.ts:97:23)
  actual:
   '[\n  {\n    "id": "cardinality_model_plot_high"\n    "modelPlotCardinality": 4675\n  }\n  {\n    "fieldName": "order_id"\n    "id": "cardinality_partition_field"\n  }\n]',
  expected:
   '[\n  {\n    "id": "cardinality_model_plot_high"\n    "modelPlotCardinality": 4711\n  }\n  {\n    "fieldName": "order_id"\n    "id": "cardinality_partition_field"\n  }\n]',
  showDiff: true }
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/4c3996242651e19ece5c0ac291743f3d012b24de) | [after](https://github.com/elastic/kibana/commit/396135a05d00127bfd459cc50ad5b592d8b00027) | diff |
| --- | --- | --- | --- |
| `ml` | 11.3MB | 11.3MB | +5.9KB |

### History
* :green_heart: [Build #81500](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/) succeeded 6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/","number":"81500","commit":"6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/","number":"81642","commit":"4c3996242651e19ece5c0ac291743f3d012b24de"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81694/)
* Commit: bfebe8fa22a9327622f7e52fefe384a353e2ae3a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/bfebe8fa22a9327622f7e52fefe384a353e2ae3a) | [after](https://github.com/elastic/kibana/commit/9b5a65a5f1bd0e95f137c906655f037a2d81725d) | diff |
| --- | --- | --- | --- |
| `ml` | 11.3MB | 11.3MB | +5.9KB |

### History
* :broken_heart: [Build #81642](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/) failed 4c3996242651e19ece5c0ac291743f3d012b24de
* :green_heart: [Build #81500](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/) succeeded 6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81500/","number":"81500","commit":"6ca5c8a6f60f2ec7d138821d4d6c30ca968ba94f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81642/","number":"81642","commit":"4c3996242651e19ece5c0ac291743f3d012b24de"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81694/","number":"81694","commit":"bfebe8fa22a9327622f7e52fefe384a353e2ae3a"}]}
PIPELINE-->
